src/hip_memory.cpp

```diff
@@ -204,6 +204,12 @@ namespace {
 inline
 void do_copy(void* __restrict dst, const void* __restrict src, size_t n,
              hsa_agent_t da, hsa_agent_t sa) {
+    if (da.handle == 0 || sa.handle == 0) { // Handle degenerate case.
+        throwing_result_check(hsa_memory_copy(dst, src, n),
+                              __FILE__, __func__, __LINE__);        
+        return;
+    }
```

Comment:
Not sure if we still need this change as we have observed hangs in hsa_memory_copy().

Reply:
Ignore the comment missed your latest commit :)